src/Capability/Constraints.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | This module defines helper types and type families for working with sets of
+-- capabilities.
+
+module Capability.Constraints
+  ( All
+  , Constraint
+  , Dict(..)
+  , None
+  ) where
+
+import Data.Kind (Constraint)
+
+-- | Evidence of a constraint that can be passed around as a value. Matching on
+-- the constructor brings the constraint into scope.
+--
+-- This is similar to the @Dict@ datatype defined in the @constraints@ package,
+-- but redefined here to avoid the dependence.
```

Comment:
Is that worth it? According to `nix-shell --pure --run 'ghc-pkg list'` `constraints` is already part of the transitive closure of dependencies. Or are there other differences I'm missing?

Reply:
Fair enough, I now just re-export it.